src/main/java/org/jabref/gui/filelist/LinkedFilesEditDialogViewModel.java

```diff
@@ -0,0 +1,126 @@
+        setValues(linkedFile);
+    }
+
+    private void checkExtension() {
+        if (!linkProperty.getValueSafe().isEmpty()) {
+
+            // Check if this looks like a remote link:
+            if (REMOTE_LINK_PATTERN.matcher(linkProperty.get()).matches()) {
+                ExternalFileTypes.getInstance().getExternalFileTypeByExt("html").ifPresent(selectedExternalFileType::setValue);
+            }
+
+            // Try to guess the file type:
+            String theLink = linkProperty.get().trim();
+            ExternalFileTypes.getInstance().getExternalFileTypeForName(theLink).ifPresent(selectedExternalFileType::setValue);
+        }
+    }
+
+    public void openBrowseDialog() {
+        String fileText = linkProperty().get();
+
+        Optional<Path> file = FileHelper.expandFilename(database, fileText, preferences.getFileDirectoryPreferences());
+
+        Path workingDir = file.orElse(Paths.get(preferences.get(JabRefPreferences.WORKING_DIRECTORY)));
+        String fileName = Paths.get(fileText).getFileName().toString();
+
+        FileDialogConfiguration fileDialogConfiguration = new FileDialogConfiguration.Builder()
+        .withInitialDirectory(workingDir)
+        .withInitialFileName(fileName)
+        .build();
+
+        dialogService.showFileOpenDialog(fileDialogConfiguration).ifPresent(path -> {
+            // Store the directory for next time:
+            preferences.put(JabRefPreferences.WORKING_DIRECTORY, path.toString());
+
+            // If the file is below the file directory, make the path relative:
+            List<Path> fileDirectories = database.getFileDirectoriesAsPaths(preferences.getFileDirectoryPreferences());
+            path = FileUtil.shortenFileName(path, fileDirectories);
+
+            linkProperty().set(path.toString());
+            checkExtension();
+        });
+    }
+
+    //
+    public void setValues(LinkedFile entry) {
+        descriptionProperty.set(entry.getDescription());
+        linkProperty.set(entry.getLink());
+
+        selectedExternalFileType.setValue(null);
+
+        // See what is a reasonable selection for the type combobox:
+        Optional<ExternalFileType> fileType = ExternalFileTypes.getInstance().fromLinkedFile(entry, false);
+        if (fileType.isPresent() && !(fileType.get() instanceof UnknownExternalFileType)) {
+            selectedExternalFileType.setValue(fileType.get());
+        } else if ((entry.getLink() != null) && (!entry.getLink().isEmpty())) {
+            checkExtension();
+        }
+    }
+
+    public StringProperty linkProperty() {
+        return linkProperty;
+    }
+
+    public StringProperty descriptionProperty() {
+        return descriptionProperty;
+    }
+
+    public ListProperty<ExternalFileType> externalFileTypeProperty() {
+        return externalfilesTypes;
+    }
+
+    public ObjectProperty<ExternalFileType> getSelectedExternalFileType() {
```

Comment:
this method should be called `selectedExternalFileTypeProperty` to adhere to JavaFX conventions